File: course/tests/rotateImage.cpp
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

int main(void){
    Mat source, imageRotate, result;

    source = imread("dog.jpg", 1);

    Point2f center(source.cols/2, source.rows/2);
    double rotationAngle = 30;
    double scale = 1;

    imageRotate = getRotationMatrix2D(center, rotationAngle, scale);
    cout << imageRotate << endl;

    warpAffine(source, result, imageRotate, Size(source.cols, source.rows));
    
    namedWindow("Original Image", WINDOW_AUTOSIZE);
    namedWindow("Rotated Image", WINDOW_AUTOSIZE);

    imshow("Original Image", source);
    imshow("Rotated Image", result);

    waitKey(0);
}

File: course/tests_py/rotateImage.py
import cv2

source = cv2.imread('tests/dog.jpg', 1)

dim = source.shape

rotationAngle = -30
scaleFactor = 1

rotationMatriz = cv2.getRotationMatrix2D((dim[0]/2, dim[1]/2), rotationAngle, scaleFactor)

print rotationMatriz

result = cv2.warpAffine(source, rotationMatriz, (dim[1], dim[0]))

cv2.imshow("Original", source)
cv2.imshow("Rotated Image", result)

cv2.waitKey(0)